### PR TITLE
Update 8.0: The TEXT Split

### PR DESCRIPTION
**Pull request by HellishBro**
strings!
strings!
more strings!
string-time

### CODE REVIEW

**Review by XTerPL — CHANGES_REQUESTED** [review]
1) please make sure that old scripts work properly in this new version

example script which DOES not load:
[Long Texts.json](https://github.com/DFOnline/DFScript/files/13806985/Long.Texts.json)

2) i added '___objectType' for a reason, please use that instead of overcomplicating the ScriptValue json serialization and deserialization

**Comment by HellishBro** [comment]
where is __objectType used at

**Comment by XTerPL** [comment]
> where is __objectType used at

In the ScriptValue.Serializer.deserialize method and the ScriptDictionaryValue.Serializer.serialize method

**Comment by HellishBro** [comment]
ok, will fix tmr. pls ping on disc when comment

**Comment by HellishBro** [comment]
@XTerPL pls review again 🙏🏻

**Comment by HellishBro** [comment]
!(strings)[https://images.squarespace-cdn.com/content/v1/57f5648637c581229b2e436f/1563893351514-F4YOO2V16U4PLEKZX1P5/Screen+Shot+2019-07-22+at+12.49.28+PM.png]

**Review by TechStreetDev — CHANGES_REQUESTED** [review]
Why is there so many formatting changes... ?

**Comment by HellishBro** [comment]
i cannot undo formatting. i am bad at git. also i am debugging it and made
a lot of changes to the code

**Comment by HellishBro** [comment]
k school is starting so progress is slow. also having to solve that merge thingy aurfhfbjjb

**Comment by HellishBro** [comment]
> Why is there so many formatting changes... ?

uh idk the unformatted one gave me an eye bleed. i also dont know how to undo a commit so ig im stuck here with a lot of formatting changes and stuff

**Comment by HellishBro** [comment]
> 1) please make sure that old scripts work properly in this new version
> 
> example script which DOES not load:
> [Long Texts.json](https://github.com/DFOnline/DFScript/files/13806985/Long.Texts.json)
> 
> 2) i added '___objectType' for a reason, please use that instead of overcomplicating the ScriptValue json serialization and deserialization

also, is this the only issue rn with the code? or is there more that i need to fix

**Comment by HellishBro** [comment]
@TechStreetDev @XTerPL problems addressed lel
I fixed the unknown menu type thing, but I still cannot revert the formatting changes.
i hope its merge-able

**Review by TechStreetDev — CHANGES_REQUESTED** [review]
I will not be approving this till the formatting changes are resolved, they are completely unnecessary, mess with the codespace and prevent us from merging this easily into other versions of the mod.

You opened the pull request for text related additions so thats all I want to see when I review it.

**Comment by HellishBro** [comment]
@TechStreetDev hopefully the formatting should be fixed

**Comment by TechStreetDev** [comment]
Lots of unnecessary formatting changes are still here

**Comment by HellishBro** [comment]
> Lots of unnecessary formatting changes are still here

which ones?

**Comment by HellishBro** [comment]
@XTerPL @TechStreetDev the code should be good now :thumbsup:

**Comment by HellishBro** [comment]
bump

File: src/main/java/io/github/techstreet/dfscript/script/ScriptManager.java [hunk 1]
```diff
@@ -49,6 +48,7 @@ public class ScriptManager implements Loadable {
         .registerTypeAdapter(ScriptPart.class, new ScriptPart.Serializer())
         .registerTypeAdapter(ScriptArgument.class, new ScriptArgument.Serializer())
         .registerTypeAdapter(ScriptTextArgument.class, new ScriptTextArgument.Serializer())
+            .registerTypeAdapter(ScriptStringArgument.class, new ScriptStringArgument.Serializer())
```
Comment (SamMan56) [inline src/main/java/io/github/techstreet/dfscript/script/ScriptManager.java]: doesnt seem to be indented correctly?

File: src/main/java/io/github/techstreet/dfscript/script/ScriptManager.java [hunk 2]
```diff
@@ -207,7 +213,7 @@ private void loadScript(File file) {
             LOGGER.info("Loaded script: " + file.getName());
         } catch (Exception e) {
             LOGGER.error("Failed to load script: " + file.getName());
-//            e.printStackTrace();
+            e.printStackTrace();
```
Comment (SamMan56) [inline src/main/java/io/github/techstreet/dfscript/script/ScriptManager.java]: debug code left in?

File: src/main/java/io/github/techstreet/dfscript/script/ScriptSnippet.java [hunk 3]
```diff
@@ -22,6 +22,7 @@
 import io.github.techstreet.dfscript.script.render.ScriptPartRender;
 import io.github.techstreet.dfscript.script.repetitions.ScriptBuiltinRepetition;
 import io.github.techstreet.dfscript.script.repetitions.ScriptRepetitionType;
+import io.github.techstreet.dfscript.script.values.ScriptValue;
```
Comment (SamMan56) [inline src/main/java/io/github/techstreet/dfscript/script/ScriptSnippet.java]: ?

File: src/main/java/io/github/techstreet/dfscript/script/action/ScriptActionArgument.java [hunk 4]
```diff
@@ -55,10 +53,10 @@ public ScriptActionArgument plural(boolean plural) {
     }
 
     public ScriptActionArgument defaultValue(ScriptValue value) {
-        if(value instanceof ScriptVariable) return this;
+        if (value instanceof ScriptVariable) return this;
```
Comment (SamMan56) [inline src/main/java/io/github/techstreet/dfscript/script/action/ScriptActionArgument.java]: remaining random formatting changes

File: src/main/java/io/github/techstreet/dfscript/script/action/ScriptFunctionCall.java [hunk 5]
```diff
@@ -52,18 +53,14 @@ public void run(ScriptTask task) {
     }
 
     public ScriptFunction getFunction() {
+        DFScript.LOGGER.info("Calling function " + function);
```
Comment (SamMan56) [inline src/main/java/io/github/techstreet/dfscript/script/action/ScriptFunctionCall.java]: leftover debug?

File: src/main/java/io/github/techstreet/dfscript/script/argument/ScriptTextArgument.java [hunk 6]
```diff
@@ -4,9 +4,7 @@
 import com.google.gson.JsonObject;
 import com.google.gson.JsonSerializationContext;
 import com.google.gson.JsonSerializer;
-import io.github.techstreet.dfscript.event.system.Event;
 import io.github.techstreet.dfscript.script.action.ScriptActionArgument.ScriptActionArgumentType;
-import io.github.techstreet.dfscript.script.execution.ScriptContext;
```
Comment (SamMan56) [inline src/main/java/io/github/techstreet/dfscript/script/argument/ScriptTextArgument.java:9]: ?

File: src/main/java/io/github/techstreet/dfscript/script/conditions/ScriptConditionType.java [hunk 7]
```diff
@@ -117,52 +113,52 @@ public enum ScriptConditionType {
     IF_LIST_CONTAINS(builder -> builder.name("List Contains")
         .description("Checks if a list contains a value.")
         .icon(Items.BOOKSHELF)
-        .category(ScriptActionCategory.LISTS)
+        .category(ScriptActionCategory.CONDITIONS)
```
Comment (SamMan56) [inline src/main/java/io/github/techstreet/dfscript/script/conditions/ScriptConditionType.java:116]: is this the time for a category re-organize?

File: src/main/java/io/github/techstreet/dfscript/script/execution/ScriptActionContext.java [hunk 8]
```diff
@@ -1,5 +1,6 @@
 package io.github.techstreet.dfscript.script.execution;
 
+import io.github.techstreet.dfscript.DFScript;
```
Comment (SamMan56) [inline src/main/java/io/github/techstreet/dfscript/script/execution/ScriptActionContext.java:3]: ?

File: src/main/java/io/github/techstreet/dfscript/script/execution/ScriptVariableMap.java [hunk 9]
```diff
@@ -10,7 +10,7 @@
 import java.util.stream.Collectors;
 
 public class ScriptVariableMap {
-    private final HashMap<String, ScriptVariable> variables = new HashMap<>();
+    public final HashMap<String, ScriptVariable> variables = new HashMap<>();
```
Comment (SamMan56) [inline src/main/java/io/github/techstreet/dfscript/script/execution/ScriptVariableMap.java]: uh? not sure why text split would need this, either way make a public getter

File: src/main/java/io/github/techstreet/dfscript/script/values/ScriptBoolValue.java [hunk 10]
```diff
@@ -18,7 +21,23 @@ String typeName() {
...
+            return new ScriptDictionaryValue(map);
+        } else if (type instanceof ScriptListValue) {
+            List<ScriptValue> list = List.of(this);
+            return new ScriptListValue(list);
+        } else if (type instanceof ScriptNumberValue) {
+            return new ScriptNumberValue(asNumber());
+        } else {
+            return super.convertTo(type);
+        }
+    }
```
Comment (SamMan56) [inline src/main/java/io/github/techstreet/dfscript/script/values/ScriptBoolValue.java]: what is this change? we adding bool casting at the same time?

File: src/main/java/io/github/techstreet/dfscript/script/ScriptManager.java (hunk 1 above)
Comment (HellishBro) [reply src/main/java/io/github/techstreet/dfscript/script/ScriptManager.java]: weird intellij issue :/

File: src/main/java/io/github/techstreet/dfscript/script/ScriptManager.java (hunk 2 above)
Comment (HellishBro) [reply src/main/java/io/github/techstreet/dfscript/script/ScriptManager.java]: for better error catching with logs

File: src/main/java/io/github/techstreet/dfscript/script/action/ScriptFunctionCall.java (hunk 5 above)
Comment (HellishBro) [reply src/main/java/io/github/techstreet/dfscript/script/action/ScriptFunctionCall.java]: yes i forgor

File: src/main/java/io/github/techstreet/dfscript/script/action/ScriptActionArgument.java (hunk 4 above)
Comment (HellishBro) [reply src/main/java/io/github/techstreet/dfscript/script/action/ScriptActionArgument.java]: had to mosify that file so the formatting changes remain

File: src/main/java/io/github/techstreet/dfscript/script/values/ScriptBoolValue.java (hunk 10 above)
Comment (HellishBro) [reply src/main/java/io/github/techstreet/dfscript/script/values/ScriptBoolValue.java]: sorry this is my first proper pr got a little carried away

File: src/main/java/io/github/techstreet/dfscript/script/ScriptManager.java (hunk 1 above)
Comment (SamMan56) [reply src/main/java/io/github/techstreet/dfscript/script/ScriptManager.java]: fix it

**Comment by HellishBro** [comment]
closed as df in on 1.20.4

**Pull request closed without merging**